src/configuration.h

```diff
@@ -29,118 +30,119 @@ class Configuration {
   Configuration(std::istream& input);
   // Used to accept inline construction of streams.
   Configuration(std::istream&& input) : Configuration(input) {}
+  ~Configuration();
 
   // Shared configuration.
-  const std::string& ProjectId() const {
+  NOOPT const std::string& ProjectId() const {
```

Comment:
Also comment on the NOOPTs in this file.

Reply:
Uh, apologies for jumping in here, but `[[clang:optnone]]` is not guaranteed to be honoured even by clang. If you want to force the function to never be inlined, use a different attribute. In this case, `__attribute__((noinline))` is supported by both clang and gcc.

Reply:
There's definitely something fishy on MacOS with respect to optimizing those functions. It used to crash for me with `__attribute__((noinline))`, but not with `[[clang::optnone]]`, but no longer seems to. I'll just remove it for now, and we can investigate if the crashes reoccur.